File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()==0)return 0;
        // sort(nums.begin(),nums.end());
        
        set<int>st(nums.begin(),nums.end());
        
        int i=0;
        for(auto it:st){
            nums[i] = it;
            i++;
        }
        
        int ans = 1;
        int curr=1;
        int n = nums.size();
        
        for(int i=0;i<n-1;i++){
            if(nums[i]+1 == nums[i+1]){
                // cout<<i<<endl;
                curr++;
            }
            else{
                ans = max(ans,curr);
                curr=1;
            }
        }
        
        ans = max(ans,curr);
        
        return ans;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int n = nums.length;
        
        if(n==0)return 0;
        
        SortedSet<Integer>st = new TreeSet<>();
        
        for(int it:nums){
            st.add(it);
        }
        int [] arr = new int[st.size()];
        int j=0;
        for(int it:st){
            arr[j] = it;
            // System.out.println(it);
            j++;
        }
        
        // Arrays.sort(arr);
        
        int len=1;
        int maxi=1;
        for(int i=1;i<arr.length;i++){
            if(arr[i] == (arr[i-1]+1)){
                len++;
            }
            else{
                maxi = Math.max(maxi,len);
                len=1;
            }
        }
        
        maxi = Math.max(maxi,len);
        return maxi;
    }
}